### PR TITLE
adds max steps attribute

### PR DESCRIPTION
**Pull request by tkumor3**

### CODE REVIEW

File: extensions/scarb-snforge-test-collector/src/compilation/test_collector/config.rs [hunk 1]
```diff
@@ -87,6 +88,8 @@ pub struct SingleTestConfig {
     pub fork_config: Option<RawForkConfig>,
     /// Custom fuzzing configuration
     pub fuzzer_config: Option<FuzzerConfig>,
+    /// custom max steps
```
Comment (cptartur) [inline extensions/scarb-snforge-test-collector/src/compilation/test_collector/config.rs:91]:
```suggestion
    /// Custom max steps
```

**Review by cptartur — APPROVED** [review]

File: extensions/scarb-snforge-test-collector/src/compilation/test_collector/config.rs [hunk 2]
```diff
@@ -140,6 +145,23 @@ pub fn forge_try_extract_test_config(
         None
     };
 
+    let max_steps = if let Some(attr) = max_steps_attr {
+        if attr.args.is_empty() {
+            None
+        } else {
```
Comment (piotmag769) [inline extensions/scarb-snforge-test-collector/src/compilation/test_collector/config.rs:151]: Just now realised this approach is just wrong (I know it was used above, it has to be fixed too, sooner or later). If there are no args here we should just throw an error, it is incorrect to have sth like `#[max_steps]` there (same with `#[fork]` and `#[fuzzer]`). I would leave it as it is and create an issue for that.

File: extensions/scarb-snforge-test-collector/src/compilation/test_collector/config.rs [hunk 3]
```diff
@@ -140,6 +145,23 @@ pub fn forge_try_extract_test_config(
...
+            extract_max_steps(attr, db).on_none(|| {
+                diagnostics.push(PluginDiagnostic {
+                    severity: Severity::Error,
+                    stable_ptr: attr.args_stable_ptr.untyped(),
+                    message: "Expected max steps must be <u32>"
```
Comment (piotmag769) [inline extensions/scarb-snforge-test-collector/src/compilation/test_collector/config.rs:156]: Doesn't align exactly with the other error messages

Comment (tkumor3) [reply extensions/scarb-snforge-test-collector/src/compilation/test_collector/config.rs]: @piotmag769 updated let me know what do you think?

Comment (piotmag769) [reply extensions/scarb-snforge-test-collector/src/compilation/test_collector/config.rs]: I was thinking of "Expected max steps config must be of the form `max_steps: <u32>`"

Comment (tkumor3) [reply extensions/scarb-snforge-test-collector/src/compilation/test_collector/config.rs]: Ok, but imo it sound like `#[max_steps({max_steps: 3232})]`

Comment (piotmag769) [reply extensions/scarb-snforge-test-collector/src/compilation/test_collector/config.rs]: Up to you

**Review by piotmag769 — APPROVED** [review]

**Pull request merged**